### ISSUE TITLE
test errors

### ISSUE THREAD

**Issue by zhangwenda0518** [nextgenusfs/funannotate#879]
I need your help ！
when i run it errors happened

funannotate util prot2genome -g test.softmasked.fa -p protein.evidence.fasta   -o test

[Mar 13 02:12 PM]: Mapping 1,065 proteins to genome using diamond and exonerate
[Mar 13 02:12 PM]: Found 1,505 preliminary alignments with diamond in 0:00:04 --> generated FASTA files for exonerate in 0:00:00
Traceback (most recent call last):
  File "/home/zhangwenda/sysoft/anaconda3/envs/funannotate/lib/python3.6/site-packages/funannotate/aux_scripts/funannotate-p2g.py", line 349, in <module>
    progress=args.progress)
  File "/home/zhangwenda/sysoft/anaconda3/envs/funannotate/lib/python3.6/site-packages/funannotate/library.py", line 1266, in runMultiProgress
    if x.successful():
  File "/home/zhangwenda/sysoft/anaconda3/envs/funannotate/lib/python3.6/multiprocessing/pool.py", line 631, in successful
    assert self.ready()
AssertionError

###########################################################################################
version ：

Checking dependencies for 1.8.14
-------------------------------------------------------
You are running Python v 3.6.15. Now checking python packages...
biopython: 1.79
goatools: 1.2.3
matplotlib: 3.3.4
natsort: 8.2.0
numpy: 1.19.5
pandas: 1.1.5
psutil: 5.8.0
requests: 2.28.1
scikit-learn: 0.24.2
scipy: 1.1.0
seaborn: 0.11.2
All 11 python packages installed

You are running Perl v b'5.032001'. Now checking perl modules...
Carp: 1.38
Clone: 0.46
DBD::SQLite: 1.72
DBD::mysql: 4.050
DBI: 1.643
DB_File: 1.855
Data::Dumper: 2.183
File::Basename: 2.85
File::Which: 1.24
Getopt::Long: 2.54
Hash::Merge: 0.302
JSON: 4.10
LWP::UserAgent: 6.67
Logger::Simple: 2.0
POSIX: 1.94
Parallel::ForkManager: 2.02
Pod::Usage: 1.69
Scalar::Util::Numeric: 0.40
Storable: 3.15
Text::Soundex: 3.05
Thread::Queue: 3.14
Tie::File: 1.06
URI::Escape: 5.12
YAML: 1.30
local::lib: 2.000029
threads: 2.25
threads::shared: 1.61
All 27 Perl modules installed

Checking Environmental Variables...
$FUNANNOTATE_DB=/home/zhangwenda/database/funannotate_db
$PASAHOME=/home/zhangwenda/sysoft/anaconda3/envs/funannotate/opt/pasa-2.5.2
$TRINITY_HOME=/home/zhangwenda/sysoft/anaconda3/envs/funannotate/opt/trinity-2.8.5
$EVM_HOME=/home/zhangwenda/sysoft/anaconda3/envs/funannotate/opt/evidencemodeler-1.1.1
$AUGUSTUS_CONFIG_PATH=/home/zhangwenda/sysoft/anaconda3/envs/funannotate/config/
$GENEMARK_PATH=/home/zhangwenda/biosoft/gene_prediction/gmes_linux_64_4
All 6 environmental variables are set
-------------------------------------------------------
Checking external dependencies...
PASA: 2.5.2
CodingQuarry: 2.0
Trinity: 2.8.5
augustus: 3.5.0
bamtools: bamtools 2.5.1
bedtools: bedtools v2.30.0
blat: BLAT v35
diamond: 2.1.4
emapper.py: v1.0
ete3: 3.1.1
exonerate: exonerate 2.2.0
fasta: 36.3.8g
glimmerhmm: 3.0.4
gmap: 2017-11-15
gmes_petap.pl: 4.69_lic
hisat2: 2.2.1
hmmscan: HMMER 3.3.2 (Nov 2020)
hmmsearch: HMMER 3.3.2 (Nov 2020)
java: 11.0.1
kallisto: 0.46.1
mafft: v7.515 (2023/Jan/15)
makeblastdb: makeblastdb 2.2.31+
minimap2: 2.24-r1122
pigz: 2.4
proteinortho: 6.2.0
pslCDnaFilter: no way to determine
salmon: salmon 0.14.1
samtools: samtools 1.16.1
signalp: 5.0b
snap: 2006-07-28
stringtie: 2.2.1
tRNAscan-SE: 2.0.11 (Oct 2022)
tantan: tantan 40
tbl2asn: 25.8
tblastn: tblastn 2.2.31+
trimal: trimAl v1.4.rev15 build[2013-12-17]
trimmomatic: 0.39
All 37 external dependencies are installed

**Comment by zhangwenda0518** [comment]
I've been stuck here for a long time. Could you please give me some help?

**Comment by nextgenusfs** [comment]
I can't tell from your output what the error is, I'd try to downgrade diamond.  See here https://github.com/nextgenusfs/funannotate/issues/872

**Comment by zhangwenda0518** [comment]
> 

I tried his last answer many times, but I still didn't succeed

I tried tblastn instead of diamond, but this error occurred.

`/home/zhangwenda/sysoft/anaconda3/envs/funannotate/lib/python3.6/site-packages/funannotate-1.8.14-py3.6.egg/funannotate/aux_scripts/funannotate-p2g.py -p annotate/predict_misc/proteins.combined.fa -g /home/zhangwenda/soft-test/funannotate/test-predict_c93fd7c6-4369-4af6-af0f-2d3070b3d749/annotate/predict_misc/genome.softmasked.fa  --maxintron 3000  --exonerate_pident 80 --ploidy 1 -f tblastn --tmpdir ./tmp --tblastn_out annotate/predict_misc/p2g.diamond.out  -o test`

errors
```
[Mar 21 11:22 PM]: Mapping 1,065 proteins to genome using tblastn and exonerate
Traceback (most recent call last):
  File "/home/zhangwenda/sysoft/anaconda3/envs/funannotate/lib/python3.6/site-packages/funannotate-1.8.14-py3.6.egg/funannotate/aux_scripts/funannotate-p2g.py", line 322, in <module>
    outfile.write('>{}\n{}\n'.format(x[0], lib.softwrap(str(genome_dict[x[0]].seq))))
  File "/home/zhangwenda/sysoft/anaconda3/envs/funannotate/lib/python3.6/site-packages/Bio/File.py", line 248, in __getitem__
    record = self._proxy.get(self._offsets[key])
KeyError: 'gb|CP022974.1|'
```
this file
![image](https://user-images.githubusercontent.com/77140740/226656054-f0f9538b-3ad1-4b32-b2d8-5296c3def94f.png)


how I can do ？ I've been troubled for nearly two weeks. I once ran it successfully on another server.

**Comment by zhangwenda0518** [comment]
I downgraded diamond to 2.0.8, but the error remains.

![image](https://user-images.githubusercontent.com/77140740/226657856-4104f82b-d237-400a-9f5d-d66ab3f974c2.png)

**Comment by zhangwenda0518** [comment]
The good news is that I solved this mistake.

The error message stayed in the very early stage of prediction. I tried to manually install different versions of augustus and diamond myself. tblastn still had errors in the test data. Similarly, I tried to add my transcription group data to my personal data set, and let him use tinity to assemble it again, and the errors still occurred.These eliminate the problem of software installation.
So I checked my python environment. 
In order to solve the conda installation problem of ete3, I specified ### **python=3.6** when I created the environment.
Considering that I don't need ete3, I decided to sacrifice ete3 and **upgrade python to version 3.8.15**. it also successfully run on test data.

I re-created a new environment, using only the following commands.
`mamba create -n funannotate funannotate`

I can run on my sequencing data normally. One disadvantage is that ete3 is still wrong at this time, but other functions of funannotate are enough to meet my needs.

the errors of ete3 is
![image](https://user-images.githubusercontent.com/77140740/226919439-d85dfbd1-4555-4c78-a611-baea95bde945.png)

Have you ever encountered this mistake? And how can I solve it?
